pkg/jobconfig/files_test.go

```diff
@@ -808,8 +816,216 @@ func TestInfo_ConfigMapName(t *testing.T) {
+		},
+		{
+			name: "stale generated postsubmit is pruned",
+			jobconfig: &prowconfig.JobConfig{
+				PostsubmitsStatic: map[string][]prowconfig.Postsubmit{
+					"repo": {{JobBase: prowconfig.JobBase{Labels: map[string]string{LabelProwGenGenerated: string(Generated)}}}},
+				},
+			},
+			expectedPruned:  true,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "not stale generated presubmit is kept",
+			jobconfig: &prowconfig.JobConfig{
+				PresubmitsStatic: map[string][]prowconfig.Presubmit{
+					"repo": {{JobBase: prowconfig.JobBase{Labels: map[string]string{LabelProwGenGenerated: string(NewlyGenerated)}}}},
+				},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "not stale generated postsubmit is kept",
+			jobconfig: &prowconfig.JobConfig{
+				PostsubmitsStatic: map[string][]prowconfig.Postsubmit{
+					"repo": {{JobBase: prowconfig.JobBase{Labels: map[string]string{LabelProwGenGenerated: string(NewlyGenerated)}}}},
+				},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "not generated presubmit is kept",
+			jobconfig: &prowconfig.JobConfig{
+				PresubmitsStatic: map[string][]prowconfig.Presubmit{
+					"repo": {{JobBase: prowconfig.JobBase{Name: "job"}}},
+				},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "not generated postsubmit is kept",
+			jobconfig: &prowconfig.JobConfig{
+				PostsubmitsStatic: map[string][]prowconfig.Postsubmit{
+					"repo": {{JobBase: prowconfig.JobBase{Name: "job"}}},
+				},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "periodics are kept",
+			jobconfig: &prowconfig.JobConfig{
+				Periodics: []prowconfig.Periodic{{JobBase: prowconfig.JobBase{
+					Name:   "job",
+					Labels: map[string]string{LabelProwGenGenerated: string(NewlyGenerated)}}}},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelProwGenGenerated},
+		},
+		{
+			name: "other tool's generated periodics are kept",
+			jobconfig: &prowconfig.JobConfig{
+				Periodics: []prowconfig.Periodic{{JobBase: prowconfig.JobBase{
+					Name:   "job",
+					Labels: map[string]string{LabelProwGenGenerated: "true"}}}},
+			},
+			expectedPruned:  false,
+			generatedLabels: []string{LabelClusterInitGenerated},
+		},
+		{
+			name: "periodics are pruned when any labels are included",
+			jobconfig: &prowconfig.JobConfig{
+				Periodics: []prowconfig.Periodic{{JobBase: prowconfig.JobBase{
+					Name:   "job",
+					Labels: map[string]string{LabelClusterInitGenerated: "true"}}}},
+			},
+			expectedPruned:  true,
+			generatedLabels: []string{LabelClusterInitGenerated, "some-other-label"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Expect either unchanged or empty JobConfig
```

Comment:
Not the biggest fan of this type of test code - it can hide bugs and adds complexity. Is it hard to have literal expected data?

Reply:
I just moved this test over from `prowgen_test.go`. Seems it would be trivial to make that update though.